internal/zinc-core/src/main/scala/sbt/internal/inc/IncrementalNameHashing.scala

```diff
@@ -32,8 +32,10 @@ private final class IncrementalNameHashing(log: sbt.util.Logger, options: IncOpt
       val aNameHashes = a.nameHashes
       val bNameHashes = b.nameHashes
       val modifiedNames = ModifiedNames.compareTwoNameHashes(aNameHashes, bNameHashes)
-      val apiChange = NamesChange(className, modifiedNames)
-      Some(apiChange)
+      if (modifiedNames.regularNames.nonEmpty || modifiedNames.implicitNames.nonEmpty) {
```

Comment:
But this check is very weird. You're checking against something that `NamesChange` was already checking. This is how `NamesChange` is currently defined:
```scala
final case class NamesChange(modified0: String, modifiedNames: ModifiedNames) extends APIChange(modified0) {
  assert(
    modifiedNames.regularNames.nonEmpty || modifiedNames.implicitNames.nonEmpty,
    s"Modified names for $modified0 is empty"
  )
}
```
  
What you're doing here is no different than what we were doing before. We never return `None` for the same reason that we never got an `AssertionError` before.